signer/src/bitcoin/utxo.rs

```diff
@@ -1082,115 +1077,63 @@ impl<'a> UnsignedTransaction<'a> {
-    ///  |-------|----|-----|--------|-------------|
-    ///    magic   op   N_d   bitmap   merkle root
-    /// ```
+    /// Contradictory to version 0, version 1 of the OP_RETURN output will
+    /// always return an output, even if there are no requests.
     ///
-    /// A more full description of the merkle root can be found in the
-    /// comments for [`UnsignedTransaction::withdrawal_merkle_root`]. If
-    /// there are no withdrawal requests in the input slice then the layout
-    /// looks like:
+    /// ## Wire Format
+    /// The layout of the OP_RETURN output is as follows:
     ///
     /// ```text
-    ///  0       2    3     5        21
-    ///  |-------|----|-----|--------|
-    ///    magic   op   N_d   bitmap
+    ///  0       2    3                                           X<80
+    ///  |-------|----|-------------------------------------------|
+    ///    magic   op   encoded withdrawal IDs (optional+variable)
     /// ```
     ///
-    /// In the above layout, magic is the UTF-8 encoded string "ST", op is
-    /// the version byte, and N_d is the of deposits in this transaction
-    /// encoded as a big-endian two byte unsigned integer.
+    /// In the above layout:
+    /// - magic: UTF-8 encoded string "ST" (2 bytes)
+    /// - op: version byte (1 byte)
+    /// - encoded IDs: withdrawal request IDs encoded using idpack (variable
+    ///   length)
     ///
-    /// `None` is only returned if there are no requests in the input slice.
-    fn new_op_return_output(reqs: &Requests, state: &SignerBtcState) -> Option<TxOut> {
-        let sbtc_data = Self::sbtc_data(reqs, state)?;
-        let script_pubkey = match Self::withdrawal_merkle_root(reqs) {
-            Some(merkle_root) => {
-                let mut data: [u8; 41] = [0; 41];
-                data[..21].copy_from_slice(&sbtc_data);
-                data[21..].copy_from_slice(&merkle_root);
-
-                ScriptBuf::new_op_return(data)
-            }
-            None => ScriptBuf::new_op_return(sbtc_data),
-        };
+    /// ## Returns
+    /// - `Some(TxOut)`: if there are any withdrawal requests, the OP_RETURN
+    ///    output with encoded withdrawal IDs
+    fn new_op_return_output(reqs: &Requests, state: &SignerBtcState) -> Result<TxOut, Error> {
+        // Create OP_RETURN data
+        let mut data = PushBytesBuf::with_capacity(80);
+        data.extend_from_slice(&state.magic_bytes)?;
+        data.push(OP_RETURN_VERSION)?;
+
+        // Extract all withdrawal request IDs
+        let mut withdrawal_ids: Vec<u64> = reqs
+            .iter()
+            .filter_map(|req| req.as_withdrawal().map(|w| w.request_id))
+            .collect();
 
-        let value = Amount::ZERO;
-        Some(TxOut { value, script_pubkey })
-    }
+        // If there are any withdrawal ID's, encode them and add them to the
+        // OP_RETURN data.
+        if !withdrawal_ids.is_empty() {
+            withdrawal_ids.sort_unstable();
```

Comment:
We don't want to sort here, or we risk losing the matching between withdrawal ids and outputs. It should already be sorted (somewhere else), so if it's not we want the segmenter to complain about it.

Reply:
Yeah now I get what you guys were talking about at standup -- removed in [b02d783](https://github.com/stacks-network/sbtc/pull/1512/commits/b02d7837663c4cef165471cfb46cb1c4615b92ef)